cupy/core/core.pyx

```diff
@@ -70,6 +70,29 @@ cdef inline _should_use_rop(x, y):
 
 
 cdef tuple _HANDLED_TYPES
+cdef int fp32_compute_type = FP32_COMPUTE_TBD
+
+
+cpdef set_fp32_compute_type(type):
+    global fp32_compute_type
+    cdef int compute_capability = int(device.get_compute_capability())
+    if type in (FP32_COMPUTE_TBD, FP32_COMPUTE_DEFAULT, FP32_COMPUTE_PEDANTIC):
+        fp32_compute_type = type
+    elif type == FP32_COMPUTE_FAST_TF32:
+        if compute_capability >= 80:
+            fp32_compute_type = type
+        else:
+            fp32_compute_type = FP32_COMPUTE_DEFAULT
+            warnings.warn('FP32_COMPUTE_FAST_TF32 is only available on GPUs'
```

Comment:
It's better to warn if TF32 is requested on CUDA 10.x or earlier.

Reply:
If CUDA version is 10.x or earlier, the compute capability is 7.x or lower, so you will definitely see this warning message.

Reply:
Ah ok, CC >= 8.0 hardware does not support CUDA 10.x or earlier.
https://docs.nvidia.com/deploy/cuda-compatibility/index.html#support-title

Reply:
That's right. To be more precise, you need CUDA 11.0 or higher to use a GPU with compute capability 8.0 or higher.